src/widget.rs

```diff
@@ -30,57 +30,181 @@ use std::ops::{Deref, DerefMut};
 use glazier::kurbo::{Rect, Size};
 use piet_scene::SceneBuilder;
 
+pub use self::box_constraints::BoxConstraints;
 use self::contexts::LifeCycleCx;
-pub use self::contexts::{AlignCx, CxState, EventCx, LayoutCx, PaintCx, PreparePaintCx, UpdateCx};
+pub use self::contexts::{CxState, EventCx, LayoutCx, PaintCx, UpdateCx};
 pub use self::core::Pod;
 pub(crate) use self::core::{ChangeFlags, PodFlags, WidgetState};
-pub use self::raw_event::{LifeCycle, RawEvent};
-
-use self::align::SingleAlignment;
+pub use self::raw_event::{Event, LifeCycle};
 
 /// A basic widget trait.
 pub trait Widget {
-    fn event(&mut self, cx: &mut EventCx, event: &RawEvent);
+    /// Handle an event.
+    ///
+    /// A number of different events (in the [`Event`] enum) are handled in this
+    /// method call. A widget can handle these events in a number of ways:
+    /// requesting things from the [`EventCtx`], mutating the data, or submitting
+    /// a [`Command`].
+    ///
+    /// [`Event`]: enum.Event.html
+    /// [`EventCtx`]: struct.EventCtx.html
+    /// [`Command`]: struct.Command.html
+    fn event(&mut self, cx: &mut EventCx, event: &Event);
 
-    /// Propagate a lifecycle event.
+    /// Handle a life cycle notification.
     ///
-    /// I am not convinced this needs to be distinct from `event`. For the
-    /// moment, we're following existing Druid.
+    /// This method is called to notify your widget of certain special events,
+    /// (available in the [`LifeCycle`] enum) that are generally related to
+    /// changes in the widget graph or in the state of your specific widget.
+    ///
+    /// [`LifeCycle`]: enum.LifeCycle.html
+    /// [`LifeCycleCx`]: struct.LifeCycleCx.html
+    /// [`Command`]: struct.Command.html
     fn lifecycle(&mut self, cx: &mut LifeCycleCx, event: &LifeCycle);
 
+    /// Update the widget's appearance in response to a change in the app's
+    /// [`Data`] or [`Env`].
+    ///
+    /// This method is called when requested by the view layer.
+    /// When the appearance of the widget needs to be updated in response to
+    /// these changes, you can call [`request_paint`] or [`request_layout`] on
+    /// the provided [`UpdateCtx`] to schedule calls to [`paint`] and [`layout`]
+    /// as required.
+    ///
+    /// This method may go around.
```

Comment:
"May go around"? Is this saying that the method may go away in future, like the original comment suggested, or something else?

Reply:
I meant "may go away" and typo'ed it. That said, my current thinking is to retain it, and I am also considering plumbing the AccessKit node update mechanism through the method. That will require some more careful design.

Reply:
I was expecting the accessibility stuff to be handled more like painting with a separate `ChangeFlag`, phase and trait method. I have not yet looked at AccessKit's API however.

Using `update` as a kind of environment change event could be a good idea.